modules/core/include/opencv2/core/cvdef.h

```diff
@@ -192,7 +192,7 @@ namespace cv { namespace debug_build_guard { } using namespace debug_build_guard
 #  define CV_ENABLE_UNROLLED 1
 #endif
 
-#ifdef __GNUC__
+#if defined(__GNUC__) ||  defined(__clang__)
```

Comment:
As I know, Clang mimics as GCC 4.2.1, so it defines `__GNUC__` macro.

MacOSX:

```
$ clang -v
Apple LLVM version 10.0.0 (clang-1000.11.45.2)
Target: x86_64-apple-darwin17.7.0
Thread model: posix
InstalledDir: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin
$ clang -x c /dev/null -dM -E | grep GNUC
#define __GNUC_MINOR__ 2
#define __GNUC_PATCHLEVEL__ 1
#define __GNUC_STDC_INLINE__ 1
#define __GNUC__ 4
```

Please, could you clarify what build configuration requires this change?

Reply:
I didn't know clang defines __GNUC__, I will take out my change.